presto-product-tests/src/main/java/com/facebook/presto/tests/QueryCanceller.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpClientConfig;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.Response;
+import io.airlift.http.client.ResponseHandler;
+import io.airlift.http.client.jetty.JettyHttpClient;
+
+import java.net.URI;
+
+import static io.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static io.airlift.http.client.Request.Builder.prepareDelete;
+import static io.airlift.http.client.ResponseHandlerUtils.propagate;
+import static java.lang.String.format;
+
+public class QueryCanceller
+{
+    private String queryId;
+
+    private String serverAddress;
+
+    public QueryCanceller(String queryId, String serverAddress)
+    {
+        this.queryId = queryId;
+        this.serverAddress = serverAddress;
+    }
+
+    public Response cancel()
+    {
+        URI baseUri = URI.create(format("%s", serverAddress));
+        URI uri = uriBuilderFrom(baseUri).appendPath("/v1/query").appendPath(queryId).build();
+        Request request = prepareDelete().setUri(uri).build();
+        HttpClient httpClient = new JettyHttpClient(new HttpClientConfig());
```

Comment:
Create the HttpClient at the top level and pass it into the QueryCanceller. Make sure to close the HttpClient still.